### PR TITLE
Product listing & filters

### PR DESCRIPTION
**Pull request by kshitijthatte**
* added functionality to fetch products from API
* added more products to the backend
* added filter reducer to sort and filter products, based on
  * price
  * category
  * ratings

[Products page](https://deploy-preview-2--wonderbakery.netlify.app/products) can be viewed from the `Order Now` Button on Homepage

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+        <Route path="/products" element={<ProductsPage />} />
+        <Route
+          path="*"
+          element={
+            <main className="nav-fixed-adjust" style={{ padding: "1rem" }}>
+              <p>There's nothing here!</p>
+            </main>
+          }
+        />
+      </Routes>
```
Comment (horsemaker) [inline src/App.js:22]: You could separate your Routes into a separate component.

File: src/App.js [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
 
 export default function App() {
   return (
     <>
       <Navbar />
-      <Homepage />
+      <Routes>
+        <Route path="/" element={<Homepage />} />
+        <Route path="/products" element={<ProductsPage />} />
+        <Route
```
Comment (AbhishekkGautam) [inline src/App.js:14]: kudos for nicely handling the unwanted routes visit. 👍

File: src/pages/Homepage/index.js [hunk 3]
```diff
@@ -63,7 +64,8 @@ const Homepage = () => {
           <div className="landing-actions">
             <h1>Wonder Bakery</h1>
             <p className="text-md">Fresh baked goods made with love</p>
-            <button className="btn btn-primary">Order Now</button>
+            {/* <button className="btn btn-primary">Order Now</button> */}
```
Comment (msdeshmukh009) [inline src/pages/Homepage/index.js]: You can remove this comment.

File: src/index.js [hunk 4]
```diff
@@ -12,10 +12,10 @@ ReactDOM.render(
   <React.StrictMode>
     <BrowserRouter>
       <Routes>
-        <Route path="/" element={<App />} />
-        <Route path="mock" element={<MockmanEs />} />
+        <Route path="/*" element={<App />} />
+        <Route path="/mock" element={<MockmanEs />} />
```
Comment (horsemaker) [inline src/index.js:16]: I like this implementation of Mockman! 👌

File: src/pages/Homepage/index.js (hunk 3 above)
Comment (horsemaker) [inline src/pages/Homepage/index.js]: I think you forgot to remove this comment. If it is not necessary, do remove this.

**Review by msdeshmukh009 — APPROVED** [review]
The code is looking clean and everything is working as expected good to merge.

File: src/pages/ProductsPage/FiltersSection.js [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+            }
+          />
+          <label htmlFor="1"> 1★ &amp; above</label>
+        </div>
+      </div>
+    </aside>
+  );
+};
+
+export default FiltersSection;
```
Comment (horsemaker) [inline src/pages/ProductsPage/FiltersSection.js:171]: This file seems too long, you could divide it into components!

File: src/pages/Homepage/index.js (hunk 3 above)
Comment (AbhishekkGautam) [inline src/pages/Homepage/index.js]: we should remove unwanted comments

**Review by himadri2110 — APPROVED** [review]
Looks good to go!
UI is clean. Nice work on breaking down the layout into components 👍🏻

**Review by horsemaker — APPROVED** [review]

File: src/pages/ProductsPage/FiltersSection.js [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+        <hr className="divider" />
+
+        <label htmlFor="price" className="aside-subheader text-sm">
+          <strong>Price</strong>
+        </label>
+        <input
+          type="range"
+          className="slider"
+          min="0"
+          max="500"
```
Comment (AbhishekkGautam) [inline src/pages/ProductsPage/FiltersSection.js:65]: we can also dynamically get the max price from the products array using `map` & `Math.max()`

**Review by BelideNikhil — APPROVED** [review]
Looks good to me.
Clean code and nice Products page
GTG!

**Review by AbhishekkGautam — APPROVED** [review]
everything else looks good 👍

**Pull request merged**